Userland/Libraries/LibWeb/HTML/HTMLOptionsCollection.cpp

```diff
@@ -111,6 +111,24 @@ WebIDL::ExceptionOr<void> HTMLOptionsCollection::add(HTMLOptionOrOptGroupElement
     return {};
 }
 
+// https://html.spec.whatwg.org/#dom-htmloptionscollection-remove
+void HTMLOptionsCollection::remove(size_t index)
+{
+    // 1. If the number of nodes represented by the collection is zero, return.
+    if (length() == 0)
+        return;
+
+    // 2. If index is not a number greater than or equal to 0 and less than the number of nodes represented by the collection, return.
+    if (index < 0 || index >= length())
+        return;
+
+    // 3. Let element be the indexth element in the collection.
+    auto element = m_cached_elements[index];
```

Comment:
Similar here, this looks like brittle usage of the cache. In this case, we can probably just use the `HTMLCollection::index()` helper instead for steps 1, 2, and 3.